File: src/main/java/br/com/model/entities/classes/FormaPagamento.java
package br.com.model.entities.classes;

public enum FormaPagamento {
    DEBITO, CREDITO, PARCELADO, VISTA, BOLETO;

    public static FormaPagamento novaFormaPagamento(Integer i) {
        if (i == 1) {
            return FormaPagamento.DEBITO;
        } else if (i == 2) {
            return FormaPagamento.CREDITO;
        } else if (i == 3) {
            return FormaPagamento.PARCELADO;
        } else if (i == 4) {
            return FormaPagamento.VISTA;
        } else if (i == 5) {
            return FormaPagamento.BOLETO;
        } else {
            return null;
        }
    }
}

File: src/main/java/br/com/model/FormaPagamento.kt
package br.com.model

enum class FormaPagamento {
    DEBITO, CREDITO, PARCELADO, VISTA, BOLETO;

    companion object {
        @JvmStatic
        fun novaFormaPagamento(formaPagamento: Int): FormaPagamento {
            return when (formaPagamento) {
                1 -> {
                    DEBITO
                }

                2 -> {
                    CREDITO
                }

                3 -> {
                    PARCELADO
                }

                4 -> {
                    VISTA
                }

                5 -> {
                    BOLETO
                }

                else -> {
                    println("esse número não deveria estar aqui!$formaPagamento")
                    DEBITO
                }
            }
        }
    }
}
